### PR TITLE
[#2899] Use atomic operations

### PR DESCRIPTION
**Pull request by ffenix113**
PR to review possibility to use atomic operations to improve performance.

### CODE REVIEW

File: golang/src/envx_counters/counters.go [hunk 1]
```diff
@@ -79,48 +80,52 @@ func HitDelta(name string, delta int64) {
...
-	v := gStorage[name]
-	gStorage[name] = v + delta
+	gStorage[name] = &i
 	gLock.Unlock()
 }
 
 // Increment counter value with 1.
 func Hitf(format string, args ...interface{}) {
-	HitDeltaf(format, 1, args...)
+	HitDelta(fmt.Sprintf(format, args...), 1)
```
Comment (tuxofil) [inline golang/src/envx_counters/counters.go]: Here and below in HitDeltaf you removed ``if gDisabled`` clause and now fmt.Sprintf() will be performed disregard of gDisabled value.

File: golang/src/envx_counters/counters.go [hunk 2]
```diff
@@ -79,48 +80,52 @@ func HitDelta(name string, delta int64) {
...
 // Set new value for gauge (or counter).
 func Setf(format string, value int64, args ...interface{}) {
-	if gDisabled {
-		return
-	}
-	name := fmt.Sprintf(format, args...)
-	gLock.Lock()
-	gStorage[name] = value
-	gLock.Unlock()
+	Set(fmt.Sprintf(format, args...), value)
```
Comment (tuxofil) [inline golang/src/envx_counters/counters.go]: See note about ``if gDisabled`` above

File: golang/src/envx_counters/counters.go (hunk 1 above)
Comment (ffenix113) [reply golang/src/envx_counters/counters.go]: Fair point, you are right.

File: golang/src/envx_counters/counters.go (hunk 2 above)
Comment (ffenix113) [reply golang/src/envx_counters/counters.go]: As above - you are right.

**Comment by ffenix113** [comment]
To see actual performance improvements numbers - please refer to Redmine issue 2899

**Pull request merged**